src/components/codecSupportHelper.ts

```diff
@@ -91,119 +295,285 @@ export function getMaxBitrateSupport(): number {
 }
 
 /**
- * Get the max supported video width the active Cast device supports.
- * @param deviceId - Cast device id.
- * @returns Max supported width.
+ * Tests the max resolution supported by the device of a particular codec.
+ * @param codec - The codec in question.
+ * @param profile - The profile for the codec.
+ * @param level - The level for the codec.
+ * @param bitDepth - The bit depth of the video.
+ * @returns `number` representing the maximum resolution supported.
  */
-export function getMaxWidthSupport(deviceId: DeviceIds): number {
-    switch (deviceId) {
-        case DeviceIds.ULTRA:
-        case DeviceIds.CCGTV:
-            return 3840;
-        case DeviceIds.GEN1AND2:
-        case DeviceIds.GEN3:
-            return 1920;
-        case DeviceIds.NESTHUBANDMAX:
-            return 1280;
+export function getMaxResolutionSupported(
+    codec: VideoCodec,
+    profile: string,
+    level: number,
+    bitDepth: number
+): Resolution {
+    // This function iteratively tests the maximum resolution assuming a 16:9
+    // resolution ratio. This should be a good enough approximation for most
+    // devices.
+    //
+    // In reality, some encoders may be limited by pixel count instead of
+    // resolution, but other devices may arbitrarily limit the resolution.
+
+    let maxRes = new Resolution(0, 0);
+    const mimeType = videoCodecToMimeType(codec);
+    const codecString = getCodecString(codec, profile, level, bitDepth);
+
+    // Limit the upper bound to 32K, which is more than enough.
+    while (maxRes.width < 30720) {
```

Comment:
Technically, 32K is not `30720`, it's 32768... so either the comment above is wrong or the constant here is wrong. Perhaps this 30720 is actually meant to be a some H * W calculation?

Reply:
I'm not sure this is correct. Here are the common 16:9 resolutions:

- **4K:** 3840 × 2160
- **8K:** 7680 × 4320
- **16K:** 15360 × 8640
- **32K:** 30720 × 17280 (speculated, not common)

Reply:
That's where that number comes from, gotcha... old (16-bit) programmer brain. :)

Reply:
(you are correct that `30720` is a calculation though. it's `1920 * 16`, where `1920` is the width component of 1920 × 1080)